File: winter-android-integration-test/src/main/java/io/jentz/winter/android/test/scope/ApplicationScope.java
package io.jentz.winter.android.test.scope;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@javax.inject.Scope
@Documented
@Retention(RUNTIME)
public @interface ApplicationScope {
}

File: winter/src/main/kotlin/io/jentz/winter/inject/ApplicationScope.kt
package io.jentz.winter.inject

import javax.inject.Scope

/**
 * Scope annotation for application wide available dependencies and default qualifier for root
 * [components][io.jentz.winter.Component].
 */
@Scope
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class ApplicationScope
